challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map {|s| s.capitalize}
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select{|s| s == "dog"}
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject{|s| s == "dog"}
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.take(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.find_all{ |s| (/(c|d|C|D).*/ =~ s) == 0}
```

Comment:
The regular expression can be anchored to the start of the string. That would be more clear than comparing `=~`'s return value with zero.

Also consider character groups and case-insensitivity. There's almost as much available to learn about regular expressions as all of Ruby. 😸